src/test/java/org/apache/sysds/test/functions/frame/FrameDropInvalidLengthTest.java

```diff
@@ -0,0 +1,220 @@
+//		double[][] invalidLength =  {{2,2,2,1}};
+//		runDropInvalidLenTest( invalidLength,3, LopProperties.ExecType.SPARK);
+//	}
+
+	@Test
+	public void testNoneBadColCP() {
+		double[][] invalidLength =  {{-1,20,20,-1}};
+		runDropInvalidLenTest( invalidLength,4, LopProperties.ExecType.CP);
+	}
+
+//	@Test
+//	public void testNoneBadColSP() {
+//		double[][] invalidLength =  {{-1,20,20,-1}};
+//		runDropInvalidLenTest( invalidLength,4, LopProperties.ExecType.SPARK);
+//	}
+
+	private void runDropInvalidLenTest(double[][] colInvalidLength, int test, LopProperties.ExecType et)
+	{
+		Types.ExecMode platformOld = setExecMode(et);
+		boolean oldFlag = OptimizerUtils.ALLOW_ALGEBRAIC_SIMPLIFICATION;
+		boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
+		try {
+			getAndLoadTestConfiguration(TEST_NAME);
+			String HOME = SCRIPT_DIR + TEST_DIR;
+			fullDMLScriptName = HOME + TEST_NAME + ".dml";
+			programArgs = new String[] {"-args", input("A"), input("M"),
+					String.valueOf(rows), Integer.toString(cols), output("B")};
+			FrameBlock frame1 = new FrameBlock(schemaStrings);
+			double[][] A = getRandomMatrix(rows, cols, 10, 100, 1, 2373);
+			initFrameDataString(frame1,A, schemaStrings); // initialize a frame with one column
+			FrameWriter writer = FrameWriterFactory.createFrameWriter(Types.FileFormat.CSV);
+
+			ArrayList<Integer> badIndex = getBadIndexes(rows/4);
+			int expected = 0;
+
+			switch (test) { //Double in String
+				case 1:
+					for (int i = 0; i < badIndex.size(); i++) {
+						frame1.set(badIndex.get(i),1,"This is a very long sentence that could" +
+								" count up to multiple characters");
+					}
+					expected += badIndex.size();
+				case 2:
+					for (int i = 0; i < badIndex.size(); i++) {
+						frame1.set(badIndex.get(i), 2, "This is out of length");
+					}
+					expected += badIndex.size();
+					break;
+				case 3:
+					expected += rows*cols;
+					break;
+				case 4:
+					expected += 0;
+					break;
+			}
+			// write data frame
+			writer.writeFrameToHDFS(
+					frame1.slice(0, rows - 1, 0, cols-1, new FrameBlock()),
+					input("A"), rows, schemaStrings.length);
+			// write expected feature length matrix
+			writeInputMatrixWithMTD("M", colInvalidLength, true);
+
+			runTest(true, false, null, -1);
+			// compare output
+			HashMap<MatrixValue.CellIndex, Double> dmlOut = readDMLMatrixFromHDFS("B");
+						MatrixValue.CellIndex index = dmlOut.keySet().iterator().next(); double d = dmlOut.get(index);
+						Assert.assertEquals(expected, d, 1e-5);
+		}
+		catch (Exception ex) {
+			throw new RuntimeException(ex);
+		}
+		finally {
+			rtplatform = platformOld;
+			DMLScript.USE_LOCAL_SPARK_CONFIG = sparkConfigOld;
+			OptimizerUtils.ALLOW_ALGEBRAIC_SIMPLIFICATION = oldFlag;
+			OptimizerUtils.ALLOW_AUTO_VECTORIZATION = true;
+			OptimizerUtils.ALLOW_OPERATOR_FUSION = true;
+		}
+	}
+
+	private ArrayList<Integer> getBadIndexes(int length) {
+		ArrayList<Integer> list = new ArrayList<Integer>();
```

Comment:
In general, you want to use something like this (typed without unnecessary redundancy): `ArrayList<Integer> list = new ArrayList<>();`